### PR TITLE
Add Client type

### PR DESCRIPTION
**Pull request by Czechh**
Adds client type to simplify working with Typescript.

### CODE REVIEW

**Review by softboyjimbo — APPROVED** [review]
good call

File: src/types.ts [hunk 1]
```diff
@@ -1,3 +1,11 @@
+export interface Client {
+  index: (payload: IndexInput) => Promise<object>
+  search: (payload: SearchInput) => Promise<object[]>
+  tune: (payload: TuningInput) => Promise<object>
+  getOne: (id: string) => Promise<object>
+  deleteOne: (id: string) => Promise<object>
+}
```
Comment (softboyjimbo) [inline src/types.ts:7]: are the typings on each method unnecessary now?

Comment (Czechh) [reply src/types.ts:7]: I'm working on an update for lanchainjs and I'm noticing this error:
```langchain:build: cache miss, executing ad5815d7d733bf83
docs:build: cache miss, executing 3c58095cd6bc3ef9
docs:build: [INFO] [en] Creating an optimized production build...
docs:build: ../langchain/src/retrievers/metal.ts:23:35 - error TS2339: Property 'search' does not exist on type 'typeof import("/Users/sergioprada/temp/langchainjs/langchain/node_modules/@getmetal/metal-sdk/dist/src/metal")'.
docs:build:
docs:build: 23     const res = await this.client.search({ text: query });
docs:build:                                      ~~~~~~
docs:build:
docs:build: ../```

So I figured we're missing this.

Comment (Czechh) [reply src/types.ts:7]: So I think it's necessary? I'm not super familiar with ts.

**Pull request merged**